chirp/projects/sfda/methods/nrc.py

```diff
@@ -0,0 +1,421 @@
+# coding=utf-8
+# Copyright 2022 The Chirp Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Exploiting the Intrinsic Neighborhood Structure for SFDA."""
+
+import functools
+from typing import Dict, Tuple, Type, Optional, Union
+
+from absl import logging
+from chirp.projects.sfda import adapt
+from chirp.projects.sfda import losses
+from chirp.projects.sfda import method_utils
+from chirp.projects.sfda import model_utils
+from clu import metrics as clu_metrics
+import flax.jax_utils as flax_utils
+import flax.linen as nn
+import flax
+import jax
+import jax.numpy as jnp
+import numpy as np
+from scipy import sparse
+import tensorflow as tf
+
+
+@flax.struct.dataclass
+class NRCLoss(clu_metrics.Metric):
+  """Computes NRC's loss for the standard single-label case."""
+
+  probabilities_sum: jnp.array
+  nn_loss_sum: jnp.array
+  extended_nn_loss_sum: jnp.array
+  n_samples: int
+
+  @classmethod
+  def from_model_output(
+    cls,
+    probabilities: jnp.ndarray,
+    nn_probability: jnp.ndarray,
+    extended_nn_probability: jnp.ndarray,
+    nn_weight: jnp.ndarray,
+    extended_nn_weight: jnp.ndarray,
+    **_,
+  ) -> "NRCLoss":
+    """Computes the standard extended nearest-neighbors loss.
+
+    Args:
+      probabilities (jnp.ndarray): Model's probability for the batch.
```

Comment:
Since there is already type annotation in the function definition, can you remove these from the docstring?

Reply:
Yes ! I think that's automatically added by an add-on.